CONTRIBUTING.md

```diff
@@ -1,61 +1,520 @@
+  * [Contributing via Pull Requests](#contributing-via-pull-requests)
 
-## Reporting Bugs/Feature Requests
 
-We welcome you to use the GitHub issue tracker to report bugs or suggest features.
+## Amplify History and Ethos
+AWS Amplify aims to simplify the experience of developing cloud
+integrations on Android. Amplify codifies best practices through
+programmatic interfaces that reduce the effort needed to interact with
+cloud resources.
+
+First and foremost, Amplify exposes to you _what_ things do, and then
+_how_ best to do them. The "_what_" is a functional use case, and
+"_how_" is an opinionated implementation that you can override with
+"escape hatches." This will allow you to have higher velocity and build
+better applications by focusing less on implementation choices.
+
+Secondly, Amplify is a manifestation of
+[The Rule of Least Power](https://en.wikipedia.org/wiki/Rule_of_least_power)
+when developing against AWS. This means it encourages architectural and
+programmatic best practices that enable you to start quickly.  This
+shows by encouraging certain services (API Gateway usage vs. direct
+DynamoDB interaction) or certain connection patterns (Circuit breaker,
+retry counts and throttle up/down).
+
+__Opinionated implementations__: There are many ways to interface with AWS
+Services. Certain service interactions are favored over others. For instance, if
+sending and receiving JSON, we would prefer an API Gateway endpoint to other
+mechanisms. Amplify will programmatically help optimize for cost and
+performance through library decisions.
+
+__Declarative actions__: Amplify will provide you a reference to a generic client
+object and offer an ability to perform common actions: "RegisterUser", "Login",
+"SendObject", "UpdateObject", "StreamData". By default, you should not need to
+worry about AWS-service-specific API operations like `putItem()` with a unique
+hash -- or even which HTTP verbs to use.
+
+__Cascading service interactions__: Certain actions in a declarative style can have
+overlapping or ambiguous AWS Service implementations. With an opinionated
+implementation we can decide which Services are "primary" and which are
+"secondary" depending on what is configured. For instance sending an image will
+prefer S3 over API Gateway..
+
+__Simple, standard data objects__: Sending & Receiving data to AWS Services can
+have many parameters, which tend to show up in the SDKs. These are abstracted
+and inferred, where possible, with simple JSON that the implementation can
+reason about. Standard parameters (bucket names, stream names, partition keys,
+etc.) that are part of the implementation are extracted from a simplified
+configuration file and dynamically generated/updated in order to further allow
+focus on state and data types only.
+
+### Amplify Design
+As more and more modules are introduced to AWS Amplify, there is a clear
+necessity to modularize the library into smaller pieces so that users
```

Comment:
users -> customers

Reply:
This one I think has pros and cons. "Customers" is Amazon language, Amazon, a for-profit business entity. As a an open-source outfit, referring to our client as a customer encourages this understanding of a business relationship. However, any .org with an open-source code base would likely not call their user a customer. Apache, Linux, etc.; I stipulate that in open-source, "customer" is a weird way to talk about things.

On the other hand, we're Amazon, and that word is ingrained in our DNA. So, that reality probably wins out. Even if this phrase adds a "peculiar" twang to thangs. :-D